src/app/visit/visit.component.ts

```diff
@@ -162,6 +162,19 @@ export class VisitComponent implements OnInit, AfterViewInit, OnDestroy {
             } else {
                 this.visitPoints.push(row);
                 this.justIds = this.visitPoints.map((vp) => vp._id);
+                this.visitService
+                    .getBetterQualityOfImageFromStore(row.code)
+                    .subscribe({
+                        next: (data) => {
+                            // this.dataSource.data.filter(
+                            //     (betterImage) => betterImage.image === data,
+                            // );
+                            console.log(data);
```

Comment:
At this point you have a `data` that it should be an object with properties, _id, code, and image. This is your new image, a better quality image that you just got it from the server. All you need to do now is replace the bad quality image with this one. 
How to do it? 

first -> think where do you need to replace it what our table is binding to?
second -> how can you replace it? 
thind -> just do it and it will work 🍻

Reply:
our dataSource is what? an object? an array? string? ....